ember-file-upload/src/system/data-transfer-wrapper.ts

```diff
@@ -43,6 +47,81 @@ export default class DataTransferWrapper {
     return this.files.length ? this.files : this.items;
   }
 
+  async getFilesAndDirectories() {
```

Comment:
[Perf] All of the child functions defined within `getFilesAndDirectories` do not rely on class state or local variable closure.

Because of this, they only need to be defined once in memory, rather than every time this methodn is called.

Let's move them to module scope (you can put them in the top of this file).

I think extracting them also helps with readability of this feature.

Reply:
Moved all the functions to the top.